scripts/ci/_utils.sh

```diff
@@ -134,6 +134,14 @@ function initialize_common_environment {
             "--env" "PYTHONDONTWRITEBYTECODE" \
         )
     fi
+
+    if [[ ${CI:=} == "true" ]]; then
+        CI_CODECOV_ENV="$(bash <(curl -s https://codecov.io/env))"
```

Comment:
What crazy magic is this?! :) 

Should this be limited to just running from Github/Travis and not CI in general?

(I can tell what it's doing. Nothing to do here)

Reply:
That's standard way of using Codecov when running codecov inside docker:
https://docs.codecov.io/docs/testing-with-docker

It detects automatically which CI codecov it is running on (it supports pretty much all of the CIs ) and it return appropriate "-e ENV" sequence to pass the right variables to inside docker.

This is needed because codecov puts together code coverage from multiple jobs running in different jobs/ machines so it needs to have a way to know that they all belong to the same build. And codecove does it via CI-specific environment variables.

Previously we passed the necessary TRAVIS_ * variables in base.yaml but the way it is done now it will work well for any CI this script will run on.